packages/flutter_tools/lib/src/commands/fuchsia_reload.dart

```diff
@@ -82,61 +82,81 @@ class FuchsiaReloadCommand extends FlutterCommand {
 
   bool _list;
 
+  static final String ipv4Lookback = InternetAddress.LOOPBACK_IP_V4.address;
+
   @override
   Future<Null> runCommand() async {
     Cache.releaseLockEarly();
 
     _validateArguments();
 
     // Find the network ports used on the device by VM service instances.
-    final List<int> servicePorts = await _getServicePorts();
-    if (servicePorts.isEmpty)
+    final List<int> deviceServicePorts = await _getServicePorts();
+    if (deviceServicePorts.isEmpty)
       throwToolExit('Couldn\'t find any running Observatory instances.');
-    for (int port in servicePorts)
+    for (int port in deviceServicePorts)
       printTrace('Fuchsia service port: $port');
 
-    if (_list) {
-      await _listVMs(servicePorts);
-      return;
-    }
+    // Set up ssh tunnels to forward the device ports to local ports.
+    final List<_PortForwarder> forwardedPorts = await _forwardPorts(
+        deviceServicePorts);
+    // Wrap everything in try/finally to make sure we kill the ssh processes
+    // doing the port forwarding.
+    try {
+      final List<int> servicePorts = forwardedPorts.map(
+          (_PortForwarder pf) => pf.port).toList();
+
+      if (_list) {
+        await _listVMs(servicePorts);
+        // Port forwarding stops when the command ends. Keep the program running
+        // until directed by the user so that Observatory URLs that we print
+        // continue to work.
+        printStatus('Press any key to exit.');
```

Comment:
stdin seems to be left in line mode, so I actually have to hit Enter.

Reply:
Done